src/App.tsx

```diff
@@ -15,61 +14,159 @@ const goodsFromServer = [
   'Garlic',
 ];
 
+interface Good {
+  goodName: string,
+  id: number,
+}
+
+const preparedGoods: Good[] = goodsFromServer.map((good, index) => {
+  return {
+    goodName: good,
+    id: index,
+  };
+});
+
 enum SortType {
   NONE,
   ALPABET,
   LENGTH,
 }
 
-// Use this function in the render method
 function getReorderedGoods(
-  goods: string[],
+  goods: Good[],
   sortType: SortType,
   isReversed: boolean,
 ) {
-  // Not to mutate the original array
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // ...
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((goodA, goodB): number => {
+      const firstGood = goodA.goodName;
+      const secondGood = goodB.goodName;
+
+      switch (sortType) {
+        case SortType.ALPABET:
+          return firstGood.localeCompare(secondGood);
 
-  return visibleGoods;
+        case SortType.LENGTH:
+          return firstGood.length - secondGood.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
```

Comment:
It is good when you can avoid nested if statements


```
    switch (sortType) {
      case SortType.ALPABET:
        return visibleGoods.sort((a, b) => a.localeCompare(b));
      case SortType.LENGTH:
        return visibleGoods.sort((a, b) => a.length - b.length);
      case SortType.NONE:
      default:
        break;
    }

Reply:
I just thought not to start sorting if the state isn't changed. But I redo